src/Core/ACL.php

```diff
@@ -1,358 +1,368 @@
+		// to one recipient. By default our selector allows multiple selects amongst all contacts.
+		$sql_extra = sprintf(" AND `rel` = %d ", intval(CONTACT_IS_FRIEND));
+		$sql_extra .= sprintf(" AND `network` IN ('%s' , '%s') ", NETWORK_DFRN, NETWORK_DIASPORA);
+
+		$tabindex_attr = !empty($tabindex) ? ' tabindex="' . intval($tabindex) . '"' : '';
+
+		$hidepreselected = '';
+		if ($preselected) {
+			$sql_extra .= " AND `id` IN (" . implode(",", $preselected) . ")";
+			$hidepreselected = ' style="display: none;"';
+		}
+
+		$o .= "<select name=\"$selname\" id=\"$selclass\" class=\"$selclass\" size=\"$size\"$tabindex_attr$hidepreselected>\r\n";
+
+		$stmt = dba::p("SELECT `id`, `name`, `url`, `network` FROM `contact`
+			WHERE `uid` = ? AND NOT `self` AND NOT `blocked` AND NOT `pending` AND NOT `archive` AND `notify` != ''
+			$sql_extra
+			ORDER BY `name` ASC ", intval(local_user())
+		);
+
+		$contacts = dba::inArray($stmt);
+
+		$arr = ['contact' => $contacts, 'entry' => $o];
+
+		// e.g. 'network_pre_contact_deny', 'profile_pre_contact_allow'
+		Addon::callHooks($a->module . '_pre_' . $selname, $arr);
+
+		$receiverlist = [];
+
+		if (DBM::is_result($contacts)) {
+			foreach ($contacts as $contact) {
+				if (in_array($contact['id'], $preselected)) {
+					$selected = ' selected="selected"';
+				} else {
+					$selected = '';
+				}
+
+				$trimmed = Protocol::formatMention($contact['url'], $contact['name']);
+
+				$receiverlist[] = $trimmed;
+
+				$o .= "<option value=\"{$contact['id']}\"$selected title=\"{$contact['name']}|{$contact['url']}\" >$trimmed</option>\r\n";
+			}
+		}
+
+		$o .= '</select>' . PHP_EOL;
+
+		if ($preselected) {
+			$o .= implode(', ', $receiverlist);
+		}
+
+		Addon::callHooks($a->module . '_post_' . $selname, $o);
+
+		return $o;
+	}
+
+	private static function fixACL(&$item) {
+		$item = intval(str_replace(['<', '>'], ['', ''], $item));
+	}
+
+	/**
+	 * Return the default permission of the provided user array
+	 *
+	 * @param array $user
+	 * @return array Hash of contact id lists
+	 */
+	public static function getDefaultUserPermissions(array $user = null)
+	{
+		$matches = [];
+
+		$acl_regex = '/<([0-9]+)>/i';
+
+		preg_match_all($acl_regex, defaults($user, 'allow_cid', ''), $matches);
+		$allow_cid = $matches[1];
+		preg_match_all($acl_regex, defaults($user, 'allow_gid', ''), $matches);
+		$allow_gid = $matches[1];
+		preg_match_all($acl_regex, defaults($user, 'deny_cid', ''), $matches);
+		$deny_cid = $matches[1];
+		preg_match_all($acl_regex, defaults($user, 'deny_gid', ''), $matches);
+		$deny_gid = $matches[1];
+
+		// Reformats the ACL data so that it is accepted by the JS frontend
+                array_walk($allow_cid, 'self::fixACL');
+                array_walk($allow_gid, 'self::fixACL');
+                array_walk($deny_cid, 'self::fixACL');
+                array_walk($deny_gid, 'self::fixACL');
```

Comment:
Does this function reference notation works as well as `['self', 'fixACL']`?

Reply:
I searched the web and found this as a solution.

Reply:
Ok good!

Reply:
See here: https://stackoverflow.com/questions/3840294/how-do-i-use-a-class-method-as-a-callback-function